pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -697,6 +698,31 @@ public RebalanceResult rebalance(
     }
   }
 
+  /** Waits for jobId to be persisted using a retry policy. Tables with 100k+ segments
+    * take up to a few seconds for the jobId to persist. This ensures the jobId is present
+    * before returning the jobId to the caller, so they can correctly poll the jobId. **/
+  public void waitForJobIdToPersist(String jobId, String tableNameWithType) {
+      try {
+          // This retry policy waits at most for 3.5 to 7s in total. This is chosen to cover
+          // typical delays for tables with many segments and avoid excessive HTTP request timeouts.
+          RetryPolicies.exponentialBackoffRetryPolicy(4, 500L, 2.0)
+                  .attempt(() -> {
+            Map<String, String> controllerJobZKMetadata = getControllerJobMetadata(jobId);
+            if (controllerJobZKMetadata == null) {
+              LOGGER.warn("jobId not persisted yet while rebalancing table: {}", tableNameWithType);
+              return false;
+            }
```

Comment:
Suggest removing this warning log as it could be very common for the first attempt to fail. The warning in the exception catch should be good enough